src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,1046 @@
+            retrieveMicrophonesStatistics(statistics);
+            retrieveContentStatus(statistics);
+            retrieveConferencingCapabilities(statistics);
+            retrieveAudioStatus(statistics);
+            retrieveCollaborationStatus(statistics);
+
+            statistics.put("Audio Volume", "");
+            controls.add(createSlider("Audio Volume", 0.0f, 100.0f, Float.valueOf(retrieveVolumeLevel())));
+            statistics.put("Mute Microphones", "");
+            controls.add(createSwitch("Mute Microphones", "On", "Off", retrieveMuteStatus().equals(MuteStatus.Muted)));
+            statistics.put("Mute Local Video", "");
+            controls.add(createSwitch("Mute Local Video", "On", "Off", retrieveVideoMuteStatus()));
+            statistics.put("Reboot", "");
+            controls.add(createButton("Reboot", "Reboot", "Rebooting...", REBOOT_GRACE_PERIOD_MS));
+
+            extendedStatistics.setStatistics(statistics);
+            extendedStatistics.setControllableProperties(controls);
+
+            Integer conferenceId = retrieveActiveConferenceCallStatistics(statistics);
+            boolean validConferenceId = conferenceId != null && conferenceId > -1;
+
+            endpointStatistics.setInCall(validConferenceId);
+            if (validConferenceId) {
+                CallStats callStats = new CallStats();
+                callStats.setCallId(String.valueOf(conferenceId));
+                callStats.setProtocol(statistics.get("Active Conference#Protocol"));
+                callStats.setRequestedCallRate(DEFAULT_CALL_RATE);
+
+                AudioChannelStats audioChannelStats = new AudioChannelStats();
+                VideoChannelStats videoChannelStats = new VideoChannelStats();
+                ContentChannelStats contentChannelStats = new ContentChannelStats();
+
+                ArrayNode conferenceCallMediaStats = retrieveConferenceCallMediaStats(conferenceId);
+                retrieveConferenceCallMediaStats(conferenceCallMediaStats, audioChannelStats, videoChannelStats, callStats);
+                retrieveSharedMediaStats(contentChannelStats);
+
+                endpointStatistics.setCallStats(callStats);
+                endpointStatistics.setAudioChannelStats(audioChannelStats);
+                endpointStatistics.setVideoChannelStats(videoChannelStats);
+                endpointStatistics.setContentChannelStats(contentChannelStats);
+            }
+
+            localStatistics = extendedStatistics;
+            localEndpointStatistics = endpointStatistics;
+        } finally {
+            controlOperationsLock.unlock();
+        }
+
+        return Arrays.asList(extendedStatistics, endpointStatistics);
+    }
+
+    /**
+     * Check if the device adapter is initialized and run internalInit() if not.
+     *
+     * @throws Exception if any error occurs
+     */
+    private void checkAndInitialize() throws Exception {
+        if (!isInitialized()) {
+            internalInit();

```

Comment:
Adapter cannot initialize itself. Lifecycle if the adapter is managed by its consumer, not the adapter. If attempted to be used without initialization, must throw proper error.